### ISSUE TITLE
Include original exception in new exception in on_communication_failure

### ISSUE THREAD

**Issue by dentarg** [twingly/ecco#13]
At https://github.com/twingly/ecco/blob/v0.5.0/lib/ecco/lifecycle_failure_listener.rb#L9-L12 we get an exception from Java, and create a new Ruby exception which we return to the user with the message from the old exception.

Often the message is `nil`, so it would be useful to also include the original exception, to help the user understand what's going on.

Like Ruby does since 2.1 with [Exception Causes](http://devblog.avdi.org/2013/12/25/exception-causes-in-ruby-2-1/).

**Comment by roback** [comment]
Not sure how we should solve this. There is no `Exception#cause=` method. We could implement our own on the error classes, but would be nice if there was a way to avoid that :)

Some related comments: https://bugs.ruby-lang.org/issues/8257#note-32

**Comment by walro** [comment]
Ah yes, we could tweak the error message instead possibly. Like:

```ruby
message = "#{exception.get_name}: #{exception.get_message)"
error = Ecco::Error::CommunicationError.new(message)
```

I think Sequel does something like this when wrapping, say, PostgreSQL errors.

**Comment by dentarg** [comment]
> We could implement our own on the error classes

I was thinking that.

**Comment by walro** [comment]
RE Sequel: https://github.com/jeremyevans/sequel/blob/2241ce3835fb20a3c9d5502ba958bff7d423da39/lib/sequel/core.rb#L130-L132

They do both :) (Message and wrap).

**Comment by walro** [comment]
A downside with wrap could be that we, possibly, don't want to pass on Java objects to the end users. Kind of what we want to avoid with #26

**Comment by dentarg** [comment]
True, maybe we should just extract all info there is to our own Ruby exception

The API doesn't look that big: https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html

**Comment by dentarg** [comment]
Interesting issue about Java and Ruby exceptions: https://github.com/jruby/jruby/issues/4781

**Comment by walro** [comment]
we could do two passes:

1. just include the class name (as suggested in https://github.com/twingly/ecco/issues/13#issuecomment-386562854) (this would close this issue)
1. error super object with stuff like orignal stacktrace and what not (new issue)
